tests/fft.hpp

```diff
@@ -20,198 +20,106 @@ using DDom = ddc::DiscreteDomain<DDim...>;
-            });
-#endif
 
-    ddc::Chunk Ff_host_alloc = ddc::
-            Chunk(ddc::get_domain<DFDim<ddc::Fourier<X>>...>(Ff), ddc::HostAllocator<Tout>());
+    ddc::Chunk Ff_host_alloc(Ff.domain(), ddc::HostAllocator<Tout>());
     ddc::ChunkSpan Ff_host = Ff_host_alloc.span_view();
     ddc::deepcopy(Ff_host, Ff);
-#if 0
-    std::cout << "\n output:\n";
-    ddc::for_each(
-            ddc::policies::serial_host,
-            ddc::get_domain<DFDim<ddc::Fourier<X>>...>(Ff_host),
-            [=](DElem<DFDim<ddc::Fourier<X>>...> const e) {
-                (std::cout << ... << coordinate(ddc::select<DFDim<ddc::Fourier<X>>>(e)))
-                        << "->" << Kokkos::abs(Ff_host(e)) << " "
-                        << Kokkos::exp(
-                                   -(Kokkos::pow(coordinate(ddc::select<DFDim<ddc::Fourier<X>>>(e)), 2)
-                                     + ...)
-                                   / 2)
-                        << ", ";
-            });
-#endif
 
-    ddc::Chunk FFf_host_alloc
-            = ddc::Chunk(ddc::get_domain<DDim<X>...>(FFf), ddc::HostAllocator<Tin>());
+    ddc::Chunk FFf_host_alloc(FFf.domain(), ddc::HostAllocator<Tin>());
     ddc::ChunkSpan FFf_host = FFf_host_alloc.span_view();
     ddc::deepcopy(FFf_host, FFf);
-#if 0
-    std::cout << "\n iFFT(FFT):\n";
-    ddc::for_each(
-            ddc::policies::serial_host,
-            ddc::get_domain<DDim<X>...>(FFf_host),
-            [=](DElem<DDim<X>...> const e) {
-                (std::cout << ... << coordinate(ddc::select<DDim<X>>(e)))
-                        << "->" << Kokkos::abs(FFf_host(e)) << " " << Kokkos::abs(f_host(e))
-                        << ", ";
-            });
-#endif
 
     double criterion = Kokkos::sqrt(ddc::transform_reduce(
-            ddc::get_domain<DFDim<ddc::Fourier<X>>...>(Ff_host),
+            Ff_host.domain(),
             0.,
             ddc::reducer::sum<double>(),
             [=](DElem<DFDim<ddc::Fourier<X>>...> const e) {
-                return Kokkos::
-                               pow(Kokkos::abs(Ff_host(e))
-                                           - Kokkos::exp(
-                                                   -(Kokkos::
-                                                             pow(coordinate(ddc::select<
-                                                                            DFDim<ddc::Fourier<X>>>(
-                                                                         e)),
-                                                                 2)
-                                                     + ...)
-                                                   / 2),
-                                   2)
-                       / (LastSelector<std::size_t, X, X...>(Nx / 2, Nx) * ...);
+                double const xn2
+                        = (Kokkos::pow(ddc::coordinate(ddc::select<DFDim<ddc::Fourier<X>>>(e)), 2)
+                           + ...);
```

Comment:
In order to avoid the costly `pow` function I can suggest this trick:
```suggestion
                double tmp = 0;
                double const xn2 = ((tmp = ddc::coordinate(ddc::select<DFDim<ddc::Fourier<X>>>(e)), tmp * tmp) + ...);
```
I don't know if this is better or not

Reply:
or create a `ddc::detail::pow2(double)` function

Reply:
That is a good idea, a lambda should also do the trick!